nodebuilder/share/module.go

```diff
@@ -34,7 +34,75 @@ func ConstructModule(tp node.Type, cfg *Config, options ...fx.Option) fx.Option
 		fx.Supply(*cfg),
 		fx.Error(cfgErr),
 		fx.Options(options...),
-		fx.Provide(newModule),
+		fx.Provide(newShareModule),
+		shrexSubComponents(),
+	)
+
+	bridgeAndFullComponents := fx.Options(
+		fx.Provide(getters.NewStoreGetter),
+		discoveryComponents(cfg),
```

Comment:
Discovery is used for all types of node, so could be extracted to base components.

Reply:
done